lib/shared/rest_client_extensions.rb

```diff
@@ -1,17 +1,14 @@
 
 module RestClient::Response
   def remote_id
-    link_self = json["_links"]["self"]["href"]
-    match_data = link_self.match(/\/(\d+)$/)
-    match_data.nil? ? nil : match_data[1].to_i
+    json['id']
   end
 
-  def json
+  def json #NEED TO CHECK THIS! There is no more any remote_id
     #!! need a rescue around this
     j = JSON::parse(body)
     # Insert remote IDs
     #!! rescues, ifs etc needed
-    j['_links']['self']['remote_id'] = j["_links"]["self"]["href"]
     if j['embedded'] and !j['embedded']['remote_id']
```

Comment:
Maybe we could remove this part of the code as we may not require the remote_id anymore. Actually we can build a similar behavior if required, but considering that the current API always provide ID information, we can go without this part.